app/controllers/posts_controller.rb

```diff
@@ -282,6 +282,68 @@ def show_revision
     render nothing: true
   end
 
+  def revert
```

Comment:
This is a really long method to be in a controller. It looks fairly clean, but the test coverage is fairly low considering all the branches you have. 

At the very least I think it should have a few more tests, but any way you could break it up (Sevice Class, more methods) would be good too.